DESIGN.md

```diff
@@ -0,0 +1,149 @@
+# Design
+
+Utapi tracks metrics of a service's usage. Metrics provided by Utapi include the
+number of incoming and outgoing bytes, number of objects being stored, the
+storage used in bytes, and a count of operations performed on a service's
+resources. Operations supported by Utapi include APIs offered by Scality's [S3
+Server](https://github.com/scality/S3). Metrics can be retrieved for a given
+time range in a service's history.
+
+## Time Range
+
+Utapi offers metrics for a time range provided by the user. For example, Utapi
+allows a user to view all actions occurring over the course of a particular
+month, week, or day. Time ranges are customizable up to a precision of fifteen
+minutes.
+
+Note: A time range precision of less than fifteen minutes can be supported as a
+feature request and could be set as a configurable option.
+
+### Timestamps
+
+Metrics provided to Utapi are set to the latest fifteen minute interval (i.e.,
+00:00:00, 00:15:00, 00:30:00, or 00:45:00). For example, if a user creates a
+bucket at 06:15:01, the operation will have a timestamp of 06:15:00. All
+timestamps are then formatted as a UNIX epoch expressed in milliseconds. During
+a listing of metrics, then, we can know that this operation occurred sometime
+between 06:15:00 and 06:29:59.
+
+#### Example
+
+| current time | timestamp | UNIX epoch timestamp |
+|--------------|-----------|----------------------|
+| 06:15:01     | 06:15:00  | 1483280100000        |
+| 06:29:59     | 06:15:00  | 1483280100000        |
+| 06:31:00     | 06:30:00  | 1483281000000        |
+| 07:01:00     | 07:00:00  | 1483282800000        |
+
+![timestamp graphic](res/timestamp-graphic.png)
+
+### Data Storage
+
+Utapi uses Redis as a database for storage of its metrics values. Accordingly,
+it uses two different Redis data types: Sorted Sets and Strings. This document
+describes how these two data types are used by Utapi. For further information on
+data types see the Redis [documentation](https://redis.io/topics/data-types).
+
+#### Sorted Sets
+
+The Redis keys for number of objects and storage utilized metrics are recorded
```

Comment:
Should it be "the number of objects", since you're not actually using the name of the number of objects variable?